src/main/java/com/github/princesslana/eriscasper/action/Action.java

```diff
@@ -1,32 +1,8 @@
 package com.github.princesslana.eriscasper.action;
 
-import com.github.princesslana.eriscasper.rest.Route;
-import java.util.Optional;
-import org.immutables.value.Value;
+import io.reactivex.Completable;
 
-/** An Action is a {@link Route} with the data required to execute it. */
-@Value.Immutable
-public interface Action<I, O> {
-  /**
-   * Get the {@link Route} that this Action will call.
-   *
-   * @return the Route this Action will call
-   */
-  Route<I, O> getRoute();
+public interface Action {
```

Comment:
Is there value in having this extend Function<ActionContext, Completable>

Probably reactivex Function rather than java's function.

Reply:
We could do that, good call.